app/src/main/kotlin/com/wire/android/ui/home/conversations/usecase/GetMessagesForConversationUseCase.kt

```diff
@@ -56,10 +58,17 @@ class GetMessagesForConversationUseCase @Inject constructor(
             startingOffset = max(0, lastReadIndex - PREFETCH_DISTANCE).toLong()
         ).map { pagingData ->
             pagingData.flatMap { messageItem ->
-                observeMemberDetailsByIds(messageMapper.memberIdList(listOf(messageItem)))
-                    .mapLatest { usersList ->
-                        messageMapper.toUIMessage(usersList, messageItem)?.let { listOf(it) } ?: emptyList()
-                    }.first()
+                val listWithSender: List<User> = (messageItem as? Message)?.sender?.let { listOf(it) } ?: listOf()
+                val otherUserIdList = messageMapper.memberIdList(listOf(messageItem))
+
+                if(otherUserIdList.isNotEmpty()) {
+                    observeMemberDetailsByIds(messageMapper.memberIdList(listOf(messageItem)))
+                        .mapLatest { usersList ->
+                            messageMapper.toUIMessage(listWithSender.plus(usersList), messageItem)?.let { listOf(it) } ?: emptyList()
+                        }.first()
+                } else {
+                    messageMapper.toUIMessage(listWithSender, messageItem)?.let { listOf(it) } ?: emptyList()
+                }
```

Comment:
question: this exact piece of code appears three times in three different use cases - would it be possible to extract into a reusable function?

Reply:
extracted to use case and added test